File: www/framework/WebView.js
//-------------------------------------------------------------------------------------------------------
// Copyright (C) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
//-------------------------------------------------------------------------------------------------------

//
// A control that hosts HTML.
//
function WebView() {
    ace.UIElement.call(this, ace.valueOn({ android: "Windows.UI.Xaml.Controls.WebView", ios: "AceWebView" }));
};

// Inheritance
WebView.prototype = Object.create(ace.UIElement.prototype);

WebView.prototype.getSource = function () { return this.get("WebView.Source"); };
WebView.prototype.setSource = function (url) { this.set("WebView.Source", url); };

module.exports = WebView;

File: src/android/framework/WebView.java
//-------------------------------------------------------------------------------------------------------
// Copyright (C) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
//-------------------------------------------------------------------------------------------------------
package Windows.UI.Xaml.Controls;

import android.view.ViewGroup;
import android.widget.AbsoluteLayout;

public class WebView extends android.webkit.WebView implements IHaveProperties {

	public WebView(android.content.Context context) {
		super(context);
	}

	// IHaveProperties.setProperty
	public void setProperty(String propertyName, Object propertyValue) {
        if (propertyName.equals("WebView.Source")) {
            this.loadUrl((String)propertyValue);
        }
		else if (!ViewGroupHelper.setProperty(this, propertyName, propertyValue)) {
			throw new RuntimeException("Unhandled property for " + this.getClass().getSimpleName() + ": " + propertyName);
		}
	}
}
